mitiq/typing.py

```diff
@@ -115,11 +124,47 @@ def nqubits(self) -> int:
     def asarray(self) -> npt.NDArray[np.int64]:
         return self._bitstrings
 
-    def __getitem__(self, indices: List[int]) -> npt.NDArray[np.int64]:
-        return np.array([self._measurements[i] for i in indices]).T
-
-    def __iter__(self) -> Iterable[Bitstring]:
-        yield from self.result
+    @classmethod
+    def from_counts(
+        cls,
+        counts: Dict[str, int],
+        qubit_indices: Optional[Tuple[int, ...]] = None,
+    ) -> "MeasurementResult":
+        """Initializes a MeasurementResult from a dictionary of counts.
+
+        Example:
+        >>> MeasurementResult.from_counts({"00": 175, "11": 177})
```

Comment:
This looks a little strange on the API-doc. WDYT of these two?
```suggestion
        **Example**::

            MeasurementResult.from_counts({"00": 175, "11": 177})
```
or
```suggestion
        **Example**: ``MeasurementResult.from_counts({"00": 175, "11": 177})``
```